Gems/AWSClientAuth/Code/Source/Authorization/AWSClientAuthCognitoCachingAuthenticatedCredentialsProvider.cpp

```diff
@@ -42,12 +42,19 @@ namespace AWSClientAuth
 
         if (!identityRepository.HasIdentityId())
         {
-            auto accountId = identityRepository.GetAccountId();
-            auto identityPoolId = identityRepository.GetIdentityPoolId();
-
             Aws::CognitoIdentity::Model::GetIdRequest getIdRequest;
-            getIdRequest.SetIdentityPoolId(identityPoolId);
 
+            // Only call SetIdentityPoolId if there's actually a pool id.
+            // SetIdentityPoolId will cause AWS to think there's an id even if it's empty.
+            // This leads AWS API calls to pass back a warning about an "invalid" pool id,
+            //     rather than (properly) passing back an error about not having a pool id.
+            const Aws::String identityPoolId = identityRepository.GetIdentityPoolId();
+            if (!identityPoolId.empty())
+            {
+                getIdRequest.SetIdentityPoolId(identityPoolId);
+            }
+
+            auto accountId = identityRepository.GetAccountId();
             if (!accountId.empty()) // new check
```

Comment:
what's "new check" mean? maybe remove the comment?

Reply:
Ha! Good call... fixed